### DISCUSSION TITLE
SDK Logout does not work on mode cookie?

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#10027] [Archive]
When trying to logout things are strange.
When I am using the json mode, i can login and logout.
But using the authMode: "cookie" i can login and request resources, but i am not able to logout:

"Error: "refresh_token" is required in either the JSON payload or Cookie"

This is the code from the repo, where i believe is a problem:

```js
private static getTransport(url, storage){
		return new Transport({
			url: url,
			beforeRequest: (config) => {
				const token = storage.auth_token;
				const bearer = token
					? token.startsWith(`Bearer `)
						? String(storage.auth_token)
						: `Bearer ${storage.auth_token}`
					: '';

				return {
					...config,
					headers: {
						Authorization: bearer,
						...config.headers,
					},
				};
			}
		});
	}
```

**Comment by rijkvanzanten** [comment]
Heya! 

This sounds like a (potential) bug. Mind [opening a new issue](https://github.com/directus/directus/issues/new)? Thanks!

**Reply by mdestefano** [reply]
I believe that this is still a bug. I was running on sveltekit with mode cookie and LocalMemory. Logout function returned error since the "refresh_token" was missing. I turned to json mode and it worked.

**Reply by rijkvanzanten** [reply]
In that case, the same answer still applies 🙂 Mind opening a new issue in the appropriate repo? https://github.com/directus/sdk/issues Thanks!

**Comment by kwbtdisk** [comment]
Same here and on my case `$directusSdk.auth.resetStorage()` fixed the problem 
```typescript
await $directusSdk.auth.logout()
// @ts-ignore
$directusSdk.auth.resetStorage()
```
